src/main/scala/com.snowplowanalytics.snowplow/scalatracker/TrackerExceptionHandler.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2015-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.scalatracker
+
+import java.io.{PrintWriter, StringWriter}
+
+import org.json4s.JsonDSL._
+
+import scala.util.control.NonFatal
+
+object TrackerExceptionHandler {
+
+  /**
+    * ExceptionHandler class
+    *
+    * @param tracker Tracker
+    */
+  implicit class ExceptionHandler(tracker: Tracker) {
+
+    private val MaxMessageLength       = 2048
+    private val MaxStackLength         = 8096
+    private val MaxThreadNameLength    = 1024
+    private val MaxClassNameLength     = 1024
+    private val MaxExceptionNameLength = 1024
+
+    /**
+      * Sends a Snowplow Event when error is non fatal and then re-throws.
+      *
+      * @param error The throwable
+      */
+    def errorHandler(error: Throwable): Unit = error match {
+      case NonFatal(e) =>
```

Comment:
I'm just wondering what behavior we can expect if somebody will try to invoke tracking code after a fatal exception. Somebody already catched an exception, which can be fatal and now tries to track it and we instead just do "no op". On the other hand people want to know about their fatal exceptions not from their analytics DB.

I'm not saying we should change it, just looking for discussion.

Reply:
Well basically it is handling all error child of `Exception`. It means exclude all child of `java.lang.Error` such as `OutOfMemoryError`, `StackOverflowError`, etc. Also take a look at doc of Error class: 
"An {@code Error} is a subclass of {@code Throwable}
  that indicates serious problems that a reasonable application
  should not try to catch. Most such errors are abnormal conditions. ..."

Reply:
I don't have a strong opinion on that, but my whole point here is that we're not actually catching it. Application developer already caught it and for some reasons want to track its presence.

One particular use-case: application is running on Akka cluster in EC2 and nodes go down periodically with fatal errors due heavy load. By tracking that errors and having enabled EC2 summary context developer might learn something useful about their cluster. Maybe this is a made up example. @BenFradet do you have an opinion?